ONSdigital/ssdc-rm-ddl
# 361 DDL Managed Database Roles

# Motivation and Context

We want our core roles to be set up and managed alongside the DDL. Note that unlike the rest of the DDL, the permissions cannot easily be torn down by the rebuild script, so the roles will remain, but all their permissions will essentially be revoked by the schemas being torn down underneath them, and then re-applied with the ground zero. If the roles were given any permissions outside those schemas, those permissions would _not_ get torn down at any point in the ground zeroing.


# What has changed



- Update DB role permissions scripts
- Move DB role permissions scripts into ground zero
- Update groundzero scripts to run role definitions
- And patch to add role definitions
- Update the ddl verion

# How to test?



Try running the patch against an up to date dev environment, check the permissiosn are now present and functional.
Try rebuilding from ground zero, then checking the permissions are present and functional.

# Links



https://trello.com/c/N7EpXEGx/361-ddl-mvp-changes-for-prod-preprod-database-user-accounts-8
